coralproject/talk
# Make DSA method of redress configurable

## What does this PR do?

- Makes the DSA method of redress configurable under `Admin > Config > General > DSA`.
- Shows method of redress under the notifications tab stream side.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

Adds the configurable `MethodOfRedress` type to the `DSAConfig`.

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indices.

## How do I test this PR?

- If you previously enabled DSA, you need to modify your tenant with the following
     - The old format of:
         ```
          "dsa": {
            "enabled": true,
            "methodOfRedress": "NONE"
          },
         ```
     - Needs to be changed to:
         ```
         "dsa": {
          "enabled": true,
          "methodOfRedress": {
            "method": "NONE",
          }
        },
        ```
- Go to `Admin > Config > General > DSA`, enable DSA, and select a method of redress
    - Select and fill in fields for URL or Email if you choose them, see that they validate for a proper URL/email
- Head to notifications tab stream side
- See that there is a light blue method of redress callout above your notification items

## Where any tests migrated to React Testing Library?

No

## How do we deploy this PR?

No special considerations.
